spec/stud/interval_spec.rb

```diff
@@ -24,6 +35,59 @@
       end
     end
 
+    it 'should be able to interrupt an interval within same thread' do
+      counter = 0
+
+      Stud.interval(0.5) do
+        counter += 1
```

Comment:
+1 on probably maybe using mocks for this, and maybe even Benchmark.realtime, but that can wait until a future PR.